app/src/main/java/com/infomaniak/euria/MainActivity.kt

```diff
@@ -267,15 +270,31 @@ class MainActivity : ComponentActivity(), AppReviewManageable {
     }
 
     private fun extractFilesToShare(intent: Intent) {
-        val items = buildList {
-            intent.clipData?.let {
-                for (i in 0 until it.itemCount) {
-                    add(it.getItemAt(i).uri)
+        val uris = buildList {
+            if (intent.action == Intent.ACTION_SEND_MULTIPLE) {
+                intent.parcelableArrayList<Uri>(Intent.EXTRA_STREAM)?.let {
+                    addAll(it.toList())
+                }
+            } else if (intent.action == Intent.ACTION_SEND) {
+                var hasUrisBeenFetched = false
```

Comment:
The word "fetched" is too heavyweight for a pure in-memory check.

```suggestion
                var clipDataContainsUris = false
```